chew_test.go

```diff
@@ -0,0 +1,235 @@
+package chew
+
+import (
+	"encoding/json"
+
+	"strings"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+func TestProcessHTML(t *testing.T) {
+
+	//the goquery parser couldn't catch a broken HTML content,
+	//so the test fails because it expects an error but got none
+	brokenHTMLContent := `
+	<html><p>Unclosed tag
+	`
+
+	invalidReader := strings.NewReader(brokenHTMLContent)
+
+	_, err := processHTML(invalidReader, "invalid.html")
+	if err == nil {
+		t.Fatalf("expected an error, but got none")
+	}
```

Comment:
I would have enclosed these 2 tests with `t.Run`

Something like this (pseudo code)

```suggestion
func TestProcessHTML(t *testing.T) {

t.Run("broken HTML", func(t *testing.T) {

	//the goquery parser couldn't catch a broken HTML content,
	//so the test fails because it expects an error but got none
	brokenHTMLContent := `
	<html><p>Unclosed tag
	`

	invalidReader := strings.NewReader(brokenHTMLContent)

	_, err := processHTML(invalidReader, "invalid.html")
	if err == nil {
		t.Fatalf("expected an error, but got none")
	}
```

t.Run("valid HTML", func(t *testing.T) {
})

It would provide distinct tests and error reporting

Reply:
That's some good perspective. Thank you, @ccoVeille!